### PR TITLE
Tests for ABC Types

### LINKED ISSUE #10828: TST: add tests for ABC Types
e.g. something in ``test_common.py`` like

```
p = pd.Period(....)
self.assertTrue(isinstance(p, ABCPeriod))
```

for all of the ABC types defined.

### PR DESCRIPTION
**Pull request by DanNi0130**
closes #10828

I used self.assertIsInstance(obj, type) and inlined as many of the expressions as I could while staying PEP8 compliant. I added in the two tests that I took out, that was an accident. Let me know if this looks okay!

### CODE REVIEW

**Comment by jreback** [comment]
@DanNi0130 normally you don't need to open a new pull request, just update the existing one.

see contribution docs [here](http://pandas.pydata.org/pandas-docs/stable/contributing.html)

File: pandas/tests/test_common.py [hunk 1]
```diff
@@ -61,6 +64,38 @@ def __call__(self):
...
+    timedelta_index = pd.to_timedelta(np.arange(5), unit='s')
+    period_index = pd.period_range('2000/1/1', '2010/1/1/', freq='M')
+    categorical = pd.Categorical([1, 2, 3], categories=[2, 3, 1])
+    categorical_df = pd.DataFrame({"values": [1, 2, 3]}, index=categorical)
+    df = pd.DataFrame({'names': ['a', 'b', 'c']}, index=multi_index)
+    sparse_series = pd.Series([1, 2, 3]).to_sparse()
+    sparse_array = pd.SparseArray(np.random.randn(10))
+
+    def test_abc_indices(self):
+        self.assertIsInstance(pd.Index(['a', 'b', 'c']), com.ABCIndex)
```
Comment (jreback) [inline pandas/tests/test_common.py]: add the issue number as a comment here

Comment (jreback) [inline pandas/tests/test_common.py]: just call this ``test_abc_types`` or something; indicies are very specific

**Comment by jreback** [comment]
small comment, pls squash. ping when green.

**Comment by DanNi0130** [comment]
You want me to add the issue number as a comment in the code? I just fixed the function name.

**Comment by jreback** [comment]
yes

**Comment by DanNi0130** [comment]
Done!

**Comment by jreback** [comment]
as i said. pls squash to a single commit. ping when travis reports green.

**Comment by jreback** [comment]
merged via https://github.com/pydata/pandas/commit/18012f0d42bb46f618e2cab8ab04620a1fe92d00

thanks!

**Pull request closed without merging**